components/data-table/private/row.jsx

```diff
@@ -22,131 +23,156 @@ import {
 	DATA_TABLE_CELL,
 } from '../../../utilities/constants';
 
+import InteractiveElement from '../interactive-element';
+import CellContext from '../private/cell-context';
+import TableContext from '../private/table-context';
+import keyboardNavState from '../private/keyboard-nav-state';
+
+const InteractiveCheckbox = InteractiveElement(Checkbox);
+const InteractiveRadio = InteractiveElement(Radio);
+
+const propTypes = {
+	assistiveText: PropTypes.shape({
+		actionsHeader: PropTypes.string,
+		columnSort: PropTypes.string,
+		columnSortedAscending: PropTypes.string,
+		columnSortedDescending: PropTypes.string,
+		selectAllRows: PropTypes.string,
+		selectRow: PropTypes.string,
+	}),
+	canSelectRows: PropTypes.oneOfType([
+		PropTypes.bool,
+		PropTypes.oneOf(['checkbox', 'radio']),
+	]),
+	columns: PropTypes.arrayOf(
+		PropTypes.shape({
+			Cell: PropTypes.func,
+			props: PropTypes.object,
+		})
+	),
+	/**
+	 * Use this if you are creating an advanced table (selectable, sortable, or resizable rows)
+	 */
+	fixedLayout: PropTypes.bool,
+	id: PropTypes.string.isRequired,
+	item: PropTypes.object.isRequired,
+	onToggle: PropTypes.func,
+	rowActions: PropTypes.element,
+	selection: PropTypes.array,
+	tableId: PropTypes.string,
+};
+
 /**
  * Used internally, provides row rendering to the DataTable.
  */
-class DataTableRow extends React.Component {
-	// ### Display Name
-	// Always use the canonical component name as the React display name.
-	static displayName = DATA_TABLE_ROW;
-
-	// ### Prop Types
-	static propTypes = {
-		assistiveText: PropTypes.shape({
-			actionsHeader: PropTypes.string,
-			columnSort: PropTypes.string,
-			columnSortedAscending: PropTypes.string,
-			columnSortedDescending: PropTypes.string,
-			selectAllRows: PropTypes.string,
-			selectRow: PropTypes.string,
-		}),
-		canSelectRows: PropTypes.oneOfType([
-			PropTypes.bool,
-			PropTypes.oneOf(['checkbox', 'radio']),
-		]),
-		columns: PropTypes.arrayOf(
-			PropTypes.shape({
-				Cell: PropTypes.func,
-				props: PropTypes.object,
-			})
-		),
-		/**
-		 * Use this if you are creating an advanced table (selectable, sortable, or resizable rows)
-		 */
-		fixedLayout: PropTypes.bool,
-		id: PropTypes.string.isRequired,
-		item: PropTypes.object.isRequired,
-		onToggle: PropTypes.func,
-		rowActions: PropTypes.element,
-		selection: PropTypes.array,
-		tableId: PropTypes.string,
+const DataTableRow = (props) => {
```

Comment:
I converted this to a function component so that the I could use the slightly more concise `useContext` syntax.